### ISSUE TITLE
Improve handling of shortcodes in Post Preview

### ISSUE THREAD

**Issue by aerych** [wordpress-mobile/WordPress-iOS#6409]
A post/page preview shown by the WPWebViewController can either display the remote version of the post or page, or just the content locally cached on the user's device.
In the case of the local preview, short codes are currently not expanded, and the preview just shows the shortcode markup. 
Let's add basic support for expanding certain common short codes.  Captions and videos are good candidates since the short code should contain all the relevant information needed. We can try to add support for gallery codes but this might be limited to which images are cached on the device.  At a minimum we could show an offline placeholder for the gallery.
- [ ] captions
- [ ] video
- [ ] galleries

**Comment by rachelmcr** [comment]
Related issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/5579

**Comment by koke** [comment]
As mentioned in https://github.com/wordpress-mobile/WordPress-iOS/issues/5693#issuecomment-281732824 I think we should be able to show a real preview in most cases, and that's what we should focus on. If we can implement all that, the only time we'd show a "simple" preview is:
- On changes to a published post on a self hosted site without Jetpack
- If something fails with a real preview (changes couldn't save, authentication issues, ...)

In the first case, we should try to make that work (maybe extend WP-API so it supports autosaves).

In the meantime I'd love to improve the simple preview if it was somewhat easy to do. `caption` seems straightforward, `video` is easy if it was uploaded from the app and we have the VideoPress ID, `gallery` is only easy if the shortcode includes IDs and we have all the media locally. Since I don't think have support for galleries in the app yet, I'm not sure how often we'll be able to figure out what the shortcode should show.

In summary, if we could score some easy improvements for the simple preview I'd definitely do them, since fixing real previews properly will take some time. But the end goal should be to get rid of them completely. Now that we have a visual editor, a fake preview doesn't provide any real value. It's probably better to say when we don't support a preview than fake it.

**Comment by koke** [comment]
There were a bunch of similar issues related to previews, I did some cleanup and created a master issue with specific steps. Closing this in favor of #9282.
